src/quick-lint-js/fe/parse-statement.cpp

```diff
@@ -1651,6 +1656,19 @@ parser::parse_and_visit_function_parameters(
     });
     return function_parameter_parse_result::missing_parameters_ignore_body;
 
+    // function async f() {}  // Invalid. Should be async function f() {}
+  case token_type::identifier:
+    if (name->string_view() ==
+        u8"async"_sv) {  // In this context async is of type identifier and not
+                         // kw_async so we need to look at the name.
+      this->diag_reporter_->report(
+          diag_function_async_method{this->peek().span()});
+      this->skip();
+      return this->parse_and_visit_function_parameters(v, name);
+    }
+    [[fallthrough]];  // Should still call QLJS_PARSER_UNIMPLEMENTED if the
+                      // identifier is not "async"
```

Comment:
Minor: Instead of falling through here, we should copy-paste the code:
```suggestion
    QLJS_PARSER_UNIMPLEMENTED();
    return function_parameter_parse_result::parsed_parameters;
```

Reply:
Done. Why is it better to do it that way instead of falling through?

Reply:
When we end up writing a good diagnostic, the code will almost certainly be different between the two. The `case` code and the `default` code do not need to be kept in sync. They can evolve independently.

A fallthrough also makes it harder to add more `case`s later.